File: C++/116_Populating Next Right Pointers in Each Node.cpp
class Solution {
public:
    void connect(TreeLinkNode *root) {
        if (root == NULL)
            return;
        TreeLinkNode *now;
        while (root->left != NULL) {
            now = root;
            while (now != NULL) {
                now->left->next = now->right;
                if (now->next != NULL)
                    now->right->next = now->next->left;
                now = now->next;
            }
            root = root->left;
        }
    }
};

File: Python/116_Populating Next Right Pointers in Each Node.py
class Solution:
    # @param root, a tree link node
    # @return nothing
    def connect(self, root):
        if not root:
            return None
        while root.left:
            node = root
            while node:
                node.left.next = node.right
                if node.next:
                    node.right.next = node.next.left
                node = node.next
            root = root.left